addon/-private/system/store.js

```diff
@@ -1674,21 +1674,28 @@ Store = Service.extend({
     assert(`You must include an 'id' for ${modelName} in an object passed to 'push'`, data.id != null && data.id !== '');
     assert(`You tried to push data with a type '${modelName}' but no model could be found with that name.`, this._hasModelFor(modelName));
 
-    // If Ember.ENV.DS_WARN_ON_UNKNOWN_KEYS is set to true and the payload
-    // contains unknown keys, log a warning.
-
-    if (Ember.ENV.DS_WARN_ON_UNKNOWN_KEYS) {
-      var type = this.modelFor(modelName);
-      warn("The payload for '" + type.modelName + "' contains these unknown keys: " +
-        Ember.inspect(Object.keys(data).forEach((key) => {
-          return !(key === 'id' || key === 'links' || get(type, 'fields').has(key) || key.match(/Type$/));
-        })) + ". Make sure they've been defined in your model.",
-        Object.keys(data).filter((key) => {
-          return !(key === 'id' || key === 'links' || get(type, 'fields').has(key) || key.match(/Type$/));
-        }).length === 0,
-        { id: 'ds.store.unknown-keys-in-payload' }
-      );
-    }
+    runInDebug(() => {
+      // If Ember.ENV.DS_WARN_ON_UNKNOWN_KEYS is set to true and the payload
+      // contains unknown attributes or relationships, log a warning.
```

Comment:
Fun fact: All the code inside `runInDebug` get striped but this comment is preserved.